src/util-inl.h

```diff
@@ -502,6 +503,17 @@ void ArrayBufferViewContents<T, S>::Read(v8::Local<v8::ArrayBufferView> abv) {
   }
 }
 
+// ECMA262 20.1.2.5
+inline bool IsSafeJsInt(v8::Local<v8::Value> v) {
+  if (!v->IsNumber()) return false;
+  double v_d = v.As<v8::Number>()->Value();
+  if (std::isnan(v_d)) return false;
+  if (std::isinf(v_d)) return false;
+  if (std::trunc(v_d) != v_d) return false;  // not int
+  if (std::abs(v_d) <= static_cast<double>(kMaxSafeJsInteger)) return true;
```

Comment:
Nit: Simply returning the result of this condition also should work.

Reply:
Very true. This method is a verbatim port of the referenced part of ECMA-262 (https://www.ecma-international.org/ecma-262/9.0/index.html#sec-number.issafeinteger) though; I thought that would be clearer to follow exactly.